### ISSUE TITLE
[BUG]: Articles still in DB are not displayed in GUI

### ISSUE THREAD

**Issue by Ra0EL** [martinrotter/rssguard#1372]
### Brief description of the issue

Starting RSS Guard this morning, I noticed all my read articles are missing in (at least...) one feed.

I keep backups of DB, and with SQLite tool I could have run some comparisons, actually looking for these missing articles.
Well, it appears they are still in the DB, hidden, but they are not displayed anymore in the GUI, even after a quick *Search articles* on the whole account - and note that **Recycle bin** is empty also.

Now, I am not sure of the meaning of all these fields in `Messages` table, especially the difference and meanings of `is_deleted` and `is_pdeleted`. The only thing, I could have understood is that *pdeleted* stands for *permanently deleted*. And it seems to me that all Messages records having `is_deleted` and `is_pdeleted` set to **1** are definitely deleted after using the tool *Cleanup database*.

But it does not explain why I can also see plenty of records (for this feed) having `is_deleted`=`is_pdeleted`=0 not visible in RSS Guard, whether they are read or not.

And neither why SQLite query can find artciles having `is_deleted=1` and `is_pdeleted=0`, as well as `is_deleted=0` and `is_pdeleted=1`,  though none of them are visible in RSS Guard (not in the feed view, not in Recycle bin).

So, whatever, it seems my database is corrupted or somehow.
But I don't see any error message - in logs.
And have no idea how to check integrity of my database... and how to fix it !

### How to reproduce the bug?

No idea...

### What was the expected result?

Find back my articles, since they have `is_deleted` and `is_pdeleted` set to zero.

### What actually happened?

They are missing, not displayed in the Feed view, not displayed in Recycle bin, not found after a search in GUI.

### Debug log

[20240408.log](https://github.com/martinrotter/rssguard/files/14909137/20240408.log)

### Operating system and version

* OS: Windows 11 - 22H2 - Build 22621.3007
* RSS Guard version:  4.6.3

**Comment by martinrotter** [comment]
Hi, it depends.

is_pdeleted = 1 means article was removed from the recycle bin and should not be displayed anymore
is_deleted = 1 means article resides in recycle bin

First comes to mind, do you maybe by mistake have article list filtering enabled?

![rssguard_jq6TllEl4W](https://github.com/martinrotter/rssguard/assets/1255302/16e13e57-8aaf-4610-8b7e-978ffdc4e3b2).

Also, if you click the root "account" node in feed list, does article show up in article list?

Also, you could maybe share the DB file with me so that I could actually check by myself?

**Comment by Ra0EL** [comment]
Thank you for quick reply.
I can confirm : 
- there is no filtering
- articles do not appear when I select / set focus on : 
  - the feeds the article belongs to
  - Recycle bin
  - root node for this account. And when root node is selected, a search cannot find any of _missing_ articles, though they are still in DB with both `is_deleted` and `is_pdeleted` set to zero.

Actually, it seems that whatever the value for `is_deleted` and `is_pdeleted`, the articles are not displayed - note this only affects a limited number of feeds, not all of them.
Incidentally, I noticed that if I update these field manually with SQL request (typically set them both to zero), the articles come back in the game, visible in the GUI as expected.

Maybe my DB file is corrupted somewhere - since this seems to impact only some of the feeds. Not all of them...

And now that I am thinking about it, this issue did happen already, every now and then, in past years. But I did not really pay attention, then. Restore a backup and restart from there.

I can share my 370MB DB file with you - but I dont want to post it here 😅
